### PR TITLE
Test the remaining Controls

### PR DESCRIPTION
**Pull request by kienstra**
* This creates a Trait to more easily test controls
* It'll allow passing a config array of `$expected_settings`, to test all of the settings
* The tests I had written usually only test the first and last settings
* @todo: add tests for all of the existing controls

### CODE REVIEW

**Review by lukecarbis — DISMISSED** [review]
Nice work!

**Comment by kienstra** [comment]
This is almost ready for code review. I'll review it briefly.

File: php/admin/class-admin.php [hunk 1]
```diff
@@ -93,11 +93,11 @@ public function enqueue_scripts() {
...
 	/**
 	 * Redirect to the Settings screen if the license is being saved.
+	 *
+	 * This runs before get_current_screen() returns a value,
 	 */
 	public function maybe_settings_redirect() {
-		$page = filter_input( INPUT_GET, 'page', FILTER_SANITIZE_STRING );
-
-		if ( 'block-lab-pro' === $page ) {
+		if ( is_admin() && isset( $_GET['page'] ) && 'block-lab-pro' === $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (kienstra) [inline php/admin/class-admin.php]: This is changed because I couldn't figure out how to mock a `filter_input()` value in unit tests. Maybe there's a way.

**Comment by kienstra** [comment]
**Request For Review**

Hi @lukecarbis,
Hope you're doing well!

Could you please review this? It adds unit tests for the remaining controls.

File: README.md [hunk 2]
```diff
@@ -3,8 +3,8 @@
 Contributors: lukecarbis, ryankienstra, Stino11, rheinardkorf
 Tags: Gutenberg, Blocks
 Requires at least: 5.0
-Tested up to: 5.2.1
-Requires PHP: 5.4
+Tested up to: 5.2
+Requires PHP: 5.6
```
Comment (kienstra) [inline README.md:7]: It'll be important to note the bumped PHP version in the Changelog for the next release.

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "block-lab",
   "title": "Block Lab",
-  "version": "1.3.1",
+  "version": "1.3.2",
```
Comment (kienstra) [inline package.json]: I'm not sure why this change is showing up in this diff, but the version is already `1.3.2` in the [package.json](https://github.com/getblocklab/block-lab/blob/develop/package.json#L4) of the `develop` branch.

**Comment by kienstra** [comment]
**Ready For Review, But No Hurry**

Hi @lukecarbis,
Thanks a lot for your work on this.

No hurry, just wanted to let you know that I'm not blocking this from review anymore.

File: php/admin/class-settings.php [hunk 4]
```diff
@@ -77,21 +77,14 @@ public function register_settings() {
...
-			case 'license':
-			default:
-				include block_lab()->get_path() . 'php/views/license.php';
-				break;
-		}
-
-		$html = ob_get_clean();
-		echo '<div class="wrap block-lab-settings">' . $html . '</div>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		?>
+		<div class="wrap block-lab-settings">
```
Comment (lukecarbis) [inline php/admin/class-settings.php:81]: Good move simplifying this. Obviously the previous solution is intended to provide a way of expanding the tabs to include additional ones in the future, but we can redo that when we need it.

**Review by lukecarbis — APPROVED** [review]
✅ OK (81 tests, 1315 assertions)

File: php/admin/class-settings.php (hunk 4 above)
Comment (kienstra) [reply php/admin/class-settings.php:81]: Thanks, I wasn't quite sure if this needed simplifying. Like you mentioned, we could revert the commit and use the previous `switch` when needed.
